nucleus/tests/base-test-utils/src/main/java/org/glassfish/tests/utils/junit/BaseHK2JUnit5Extension.java

```diff
@@ -89,10 +89,10 @@
  *
  * @author David Matejcek
  */
-public class HK2JUnit5Extension
```

Comment:
I don't like renaming this class as it is not abstract and it can be still used as a standalone Hk2 Junit5 Extension. Rather the new class name should describe what it is doing ... 

`GlobalServiceLocatorResetterHk2Junit5Extension` (meh, too long) or ... maybe just `GlassFishHk2Junit5Extension` (as it is all about `Globals`). I don't know, that splitting smells a bit to me.

All those tests follow the same pattern as GlassFish - GlassFish doesn't reset the global `ServiceLocator` too. If you need it for an integration test, you can probably simply configure the failsafe plugin to fork JVM for every test class. I forgot details, perhaps I can help after some experimenting ... if you need it.
* https://maven.apache.org/surefire/maven-failsafe-plugin/integration-test-mojo.html#reuseforks
* forkCount
* forkedProcess*